src/ledger/fuzz.zig

```diff
@@ -0,0 +1,344 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+const build_options = @import("build-options");
+const ledger = @import("lib.zig");
+
+const ColumnFamily = sig.ledger.database.ColumnFamily;
+
+const allocator = std.heap.c_allocator;
+
+const Data = struct {
+    value: []const u8,
+};
+
+const cf1 = ColumnFamily{
+    .name = "data",
+    .Key = u64,
+    .Value = Data,
+};
+
+var executed_actions = std.AutoHashMap(Actions, void).init(allocator);
+
+pub const BlockstoreDB = switch (build_options.blockstore_db) {
+    .rocksdb => ledger.database.RocksDB(&.{cf1}),
+    .hashmap => ledger.database.SharedHashMapDB(&.{cf1}),
+};
+
+// Note: deleteFilesInRange is not included in the fuzzing as it is not
+// implemented in the hashmap implementation, and the RocksDB implementation
+// requires manual flushing of the memtable to disk to make the changes visible.
+const Actions = enum {
+    put,
+    get,
+    get_bytes,
+    count,
+    contains,
+    delete,
+    batch,
+};
+
+fn getKeys(map: *std.AutoHashMap(u32, Data)) !std.ArrayList(u32) {
+    var keys = std.ArrayList(u32).init(allocator);
+    var it = map.iterator();
+    while (it.next()) |entry| {
+        try keys.append(entry.key_ptr.*);
+    }
+    return keys;
+}
+
+fn createBlockstoreDB() !BlockstoreDB {
+    const ledger_path =
+        try std.fmt.allocPrint(allocator, "{s}/ledger", .{sig.FUZZ_DATA_DIR});
+
+    // ensure we start with a clean slate.
+    if (std.fs.cwd().access(ledger_path, .{})) |_| {
+        try std.fs.cwd().deleteTree(ledger_path);
+    } else |_| {}
+    try std.fs.cwd().makePath(ledger_path);
+
+    return try BlockstoreDB.open(
+        allocator,
+        .noop,
+        ledger_path,
+    );
+}
+
+pub fn run(initial_seed: u64, args: *std.process.ArgIterator) !void {
+    var seed = initial_seed;
+    const maybe_max_actions_string = args.next();
+
+    const maybe_max_actions = if (maybe_max_actions_string) |max_actions_str|
+        try std.fmt.parseInt(usize, max_actions_str, 10)
+    else
+        null;
+
+    const ledger_path =
+        try std.fmt.allocPrint(allocator, "{s}/ledger", .{sig.FUZZ_DATA_DIR});
+
+    // ensure we start with a clean slate.
+    if (std.fs.cwd().access(ledger_path, .{})) |_| {
+        try std.fs.cwd().deleteTree(ledger_path);
+    } else |_| {}
+    try std.fs.cwd().makePath(ledger_path);
+
+    var db = try createBlockstoreDB();
+
+    defer db.deinit();
+
+    var count: u64 = 0;
+
+    while (true) outer: {
+        var prng = std.Random.DefaultPrng.init(seed);
+        const random = prng.random();
+        // This is a simpler blockstore which is used to make sure
+        // the method calls being fuzzed return expected data.
+        var data_map = std.AutoHashMap(u32, Data).init(allocator);
+        defer data_map.deinit();
+        for (0..1_000) |_| {
```

Comment:
@dnut The benefit of this change is that the seed used is not just the one passed in.  The observation though is that, this fixes the iteration for each seed at 1000. - which may seem arbitrary. 

It would probably be an improvement to have this configurable (which will differ from the other fuzzer) or

Do not change the seed value like this put have a random value generated by CI.